### ISSUE TITLE
blockreassurance: Bloated code

### ISSUE THREAD

**Issue by dennispw** [PrestaShop/PrestaShop#15869]
Similar concerns as this ticket: https://github.com/PrestaShop/PrestaShop/issues/15495

The new update is good, multiple icon selection and even better, **SVG** support. Although I'm yet again concerned with the current approach.

Right now, there's an option to upload/use and SVG file, which is great. Smaller filesize and the ability to actually change the icon color _only_. But as it's done, there's an AJAX request _for every item_. Depending on the module settings when visiting a product page, **__by default__** in a "worse case" scenario this could very well mean that there would be **__9__** seperate AJAX request just to change the color of the icons: 3 icons, in the header, on the actual product page and in the footer.

Wouldn't it again be better if this was all done when uploading/saving the image and editing the contents of the file to get the proper color in instead of injecting a bunch of AJAX requests? I'm just thinking out loud here.

**Comment by khouloudbelguith** [comment]
Hi @dennispw,

Thanks for your report.
Ping @Progi1984 @PrestaShop/prestashop-core-developers what do you think?

Thanks!

**Comment by matks** [comment]
@dennispw hi ! thanks for the feedback, it's very interesting (just like the other one 😉, your comments are always neat)

I'm afraid I have quite the same answer than in the previous ticket though 😅 : this is an improvement that needs to be added to the module. I'd be happy to make these improvements but I don't have the time right now as I'm working on other topics, but if someone wants to change this behavior I'll be happy to review his/her PR.

Basically after the launch of this module in v4.1, we've got a huge pack of other things to do / fix / release so we our hands are full on other topics. So we cant make the needed modifications now 😢 . Maybe some time later, when we have a little more free time.

**Comment by Progi1984** [comment]
@dennispw I have made a PR for optimizing this point : PrestaShop/blockreassurance/pull/365

**Comment by dennispw** [comment]
> @dennispw I have made a PR for optimizing this point : [PrestaShop/blockreassurance/pull/365](https://github.com/PrestaShop/blockreassurance/pull/365)

Any steps in the right direction helps!
